source/reactor/inc/reactor/EventHandlers.h

```diff
@@ -0,0 +1,68 @@
+#ifndef Reactor_EventHandlers_H
+#define Reactor_EventHandlers_H
+
+#include <functional>
+
+namespace u {
+namespace reactor {
+class EventLoop;
+
+/**
+ * @brief Event handler function definition.
+ *
+ * Parameters to the event handler function are:
+ * - The file descriptor of the OS resource where the event was originated at. Event handler can
+ * perform, for example, accept, read, write and close operations on the file descriptor.
+ * - The event loop where it was caught. Event handler can install new handlers, remove existing
+ * handlers and even ask for loop termination.
+ */
+using EventHandler = std::function<void(int, EventLoop *)>;
```

Comment:
Both the int (that should be a `class FD` as @arnout correctly commented) and EventLoop are the _context_ of the function call, the data to be processed. 
Personally I prefer having the context grouped together and not spread as function parameters list. It does lots of good things and gives flexibility.
My OO approach would be therefore to write this class in this way:
```cpp
struct EventContext{
int fd;
EventLoop& originator;
// potentially add more context in the future
};
class EventsHandler{
void on_read(const &EventContext) = 0;
void on_write(const &EventContext) = 0;
// etc.
};
```

If we give up the idea of _replacing_ handlers on the fly, then my template based approach would be:
```cpp
template<class ConcreteEventsHandler> 
class EventLoop{
/* everything */
ConcreteEventsHandler _installedHandler_
};
```